### PR TITLE
Zep Editor Lua Integration

### PR DESCRIPTION
**Pull request by brainiac**
C++ refactoring and Lua integration of console completed. Still working on text editor lua component.

### CODE REVIEW

**Comment by brainiac** [comment]
Work-in-progress definitions: https://github.com/macroquest/mq-definitions/pull/87

**Comment by brainiac** [comment]
updated luaconsole gist: https://gist.github.com/brainiac/78171dd9c57ff0046e35f3dbebae1190

File: contrib/zep/src/window.cpp [hunk 1]
```diff
@@ -164,10 +162,19 @@ void ZepWindow::UpdateAirline()
...
+        }
+        else
+        {
+            m_airline.rightBoxes.push_back(
+                AirBox{ fmt::format("{}:{}", cursor.x + 1, cursor.y + 1), m_pBuffer->GetTheme().GetColor(ThemeColor::TabActive) });
+        }
 
 #ifdef _DEBUG
-        m_airline.leftBoxes.push_back(AirBox{ "(" + std::to_string(GetEditor().GetDisplay().GetPixelScale().x) + "," + std::to_string(GetEditor().GetDisplay().GetPixelScale().y) + ")", m_pBuffer->GetTheme().GetColor(ThemeColor::Error) });
+        //m_airline.leftBoxes.push_back(AirBox{ "(" + std::to_string(GetEditor().GetDisplay().GetPixelScale().x) + "," + std::to_string(GetEditor().GetDisplay().GetPixelScale().y) + ")", m_pBuffer->GetTheme().GetColor(ThemeColor::Error) });
```
Comment (dannuic) [inline contrib/zep/src/window.cpp:177]: should this remain in for debug builds? it looks useful

Comment (brainiac) [reply contrib/zep/src/window.cpp:177]: We don't change pixel scale, so it is always 1.0

**Pull request merged**